src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -143,6 +143,17 @@ pub struct GlobalId<'tcx> {
     pub promoted: Option<mir::Promoted>,
 }
 
+impl GlobalId<'tcx> {
+    pub fn display(self, tcx: TyCtxt<'tcx>) -> String {
```

Comment:
Would it make sense to make this the `Debug`/`Display` impl for `GlobalId` in general?

Reply:
@oli-obk  pointed me out that we cannot have `tcx` in `Debug`/`Display`

Reply:
Ah... I think some impls do that anyway via `with_tcx` or whatever it is called, but I am not sure how accepted it is to add more uses of that method.^^

Reply:
Since all callers to this method have a `tcx` available, I would punt the change to using the `tcx` from thread local storage until we have a caller that doesn't have it